tests/checkpointing/test_model_checkpoint.py

```diff
@@ -1247,6 +1247,40 @@ def on_load_checkpoint(self, *args, **kwargs):
     make_assertions(cb_restore, written_ckpt)
 
 
+def test_resume_training_preserves_old_ckpt_last(tmpdir):
+    """Ensures that the last saved checkpoint is not deleted from the previous folder when training is resumed from
+    the old checkpoint."""
+    model = BoringModel()
+    trainer_kwargs = {
+        "default_root_dir": tmpdir,
+        "max_epochs": 1,
+        "limit_train_batches": 3,
+        "limit_val_batches": 0,
+        "enable_model_summary": False,
+        "logger": False,
+    }
+    mc_kwargs = {
+        "filename": "{step}",
+        "monitor": "step",
+        "mode": "max",
+        "save_last": True,
+        "save_top_k": 2,
+        "every_n_train_steps": 1,
+    }
+    trainer = Trainer(**trainer_kwargs, callbacks=ModelCheckpoint(**mc_kwargs))
+    trainer.fit(model)
+    # Make sure that the last checkpoint file exists in the dirpath passed (`tmpdir`)
+    assert sorted(os.listdir(tmpdir / "checkpoints")) == sorted(["last.ckpt", "step=2.ckpt", "step=3.ckpt"])
```

Comment:
Order doesn't matter, could use a set

```suggestion
    assert set(os.listdir(tmpdir / "checkpoints")) == {"last.ckpt", "step=2.ckpt", "step=3.ckpt"}
```